### DISCUSSION TITLE
Question: What use cases have you used this Gem for?

### DISCUSSION THREAD

**Discussion by lucasklaassen** [Shopify/limiter#28] [Q&A]
Very interested to hear about when this gem has been used at Shopify and others who've used it. Limiting on the method level is very interesting to me.

Does it block the call and hang up resources until the limit has been lifted?

**Comment by sbfaulkner** [comment]
Hi
Yes, it blocks until the configurerd rate-limit won't be exceeded by completing the operation.

Our initial use-case was to limit the rate that we invoked a 3rd-party REST API

Taking this approach minimized changes to the utility that was running into the rate limit.

Cheers
